### DISCUSSION TITLE
In-app updates on android (if possible windows also)

### DISCUSSION THREAD

**Discussion by balaedit** [airsdk/Adobe-Runtime-Support#3094] [Ideas]
As we see that major android have in-app facility (through play store).  Is Harman Air have any implementation on this feature in future update of AIR SDK.

Android developer guide link : https://developer.android.com/guide/playcore/in-app-updates

**Comment by yuyuanhua** [comment]
You can load the swf file from ApplicationStorageDirectory and replace the old one to get auto-update, worked both on android, windows macOS. To do this, you need to write some code to manage the swf module loaded into the application domain.

**Reply by balaedit** [reply]
I don't think so but I will check. @ajwfrost Please confirm it.

**Reply by yuyuanhua** [reply]
Yes, this's what we do in out apps and it works good.

**Comment by marchbold** [comment]
Looks like something that would be possible through an ANE relatively easily. I can look into it if you want.

**Reply by balaedit** [reply]
Sure... Thanks

**Reply by marchbold** [reply]
@balaedit This functionality is actually already available through our InAppBilling ANE. You can find the documentation on it here:

https://docs.airnativeextensions.com/docs/inappbilling/in-app-updates

I will look into providing a standalone extension for this, but in the meantime you have access to it through there.
